java-checks-test-sources/default/src/main/java/checks/LazyArgEvaluationCheckSampleFPs.java

```diff
@@ -0,0 +1,61 @@
+package checks;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+
+// https://community.sonarsource.com/t/false-positive-on-java-s2629/42091
+class AFalsePositveFromTheCommunity {
+
+  void foo() {
+    final Marker myMarker = MarkerFactory.getMarker("MY_MARKER");
```

Comment:
Why not put these variables as class fields and have a single class testing the different scenarios?
The reference to the community thread can be put as a comment on top of each group of examples